utest/src-0/utest/TestSuite.scala

```diff
@@ -0,0 +1,46 @@
+package utest
+
+import utest.framework.Formatter
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import utest.EnableReflectiveInstantiation
+
+/**
+ * Marker class used to mark an `object` as something containing tests. Used
+ * for test-discovery by SBT.
+ */
+@EnableReflectiveInstantiation
+abstract class TestSuite
+  extends framework.Executor{
+  def utestFormatter: Formatter = null
+  def tests: Tests
+  @deprecated("Use `utest.Tests{...}` instead")
+  inline def apply(expr: => Unit): Tests = ${Tests.testsImpl('expr)}
```

Comment:
Would I be correct in saying that this `inline def` macro is the only thing that differs between Scala 2 and Dotty? If the rest of the code in this file is identical, we should find a way to share it: either moving this `def apply` to a version-specific `trait` that the shared `TestSuite` code can extend, or moving all the non-macro logic to a shared trait that the version-specific `TestSuite` code can extend

Reply:
Yes, this one can also be shared, I believe.